### DISCUSSION TITLE
Unveiling the new Admin Dashboard (preview)

### DISCUSSION THREAD

**Discussion by waiting-for-dev** [solidusio/solidus#5319] [Announcements]
We're releasing the first alpha version of solidus_admin. Our new admin dashboard still needs to be completed, but it's 100% usable and adds value by itself. We don't want to work on it behind closed doors and then show you what we think you want. Instead, we'd like everyone to try it out, listen to your feedback and create the solution you and your clients actually need.

In this initial launch, we present you with a fresh and revamped product listing with excellent UI and UX and new features like live search and bulk actions. You can also start exploring the new components-based architecture, making it easier to implement forward-compatible customizations.

[solidus_admin_demo.webm](https://github.com/nebulab/solidus/assets/52650/608448c7-f322-4962-88d9-b3cff7cb6d62)

## Switching Between Old and New Admin

During this development phase, where only some sections have been reworked, the new admin dashboard will transparently fall back to the old solidus_backend you are familiar with.

Operators can also explicitly opt-out by using a toggle switch on the menu's left side.

<img src="https://github.com/nebulab/solidus/assets/52650/1e632b12-b4f7-4774-94f9-6c8c1109ba8a" width="150" />

## Trying the new admin

We encourage you to check the new admin dashboard out and have your voice heard with any new ideas or concerns. You have two options:

1. Try the new admin in an isolated environment
2. Try the new admin in your store

Its source must be the [`admin/alpha/v1` branch](https://github.com/solidusio/solidus/tree/admin/alpha/v1) in the `solidusio/solidus` repository on GitHub.

### Try the new admin in an isolated environment

- Clone the Solidus repository: `git clone https://github.com/solidusio/solidus.git`.
- Navigate to the generated directory: `cd solidus`.
- Checkout the tag: `git checkout admin/alpha/v1`.
- Build and start the sandbox application: `bin/rails server`.
- Open your browse and visit `http://localhost:3000/admin`. You should see the new admin once you click on the "Products" menu item.

### Try the new admin in your store

- First, you must upgrade your store to fetch the `main` branch of the `solidusio/solidus` repository. We recommend following the [upgrade instructions](https://guides.solidus.io/upgrading-solidus/) up to Solidus v4.1 and then update your Gemfile, splitting how solidus dependencies are fetched:

```ruby
git "https://github.com/solidusio/solidus.git", branch: "admin/alpha/v1"  do
  gem "solidus_core"
  gem "solidus_api"
  gem "solidus_backend"
  gem "solidus_admin"
end
```

- Run `bundle update`
- Run `bin/rails g solidus_admin:install`
- Run `bin/rails server`
- Visit `http://localhost:3000/admin` in your browser. You should see the new admin once you click on the "Products" menu item.

🚧 **WARNING** 🚧

Notice that any customization you made on the legacy admin will not automatically be transferred to the new admin dashboard. We'll publish detailed instructions on customizing the new admin and port legacy patches as soon as possible.

🚧 **WARNING** 🚧

## Troubleshooting

If you encounter any issues during installation, please make sure to follow the instructions we provided above. 

For further assistance, we can [help troubleshoot](https://github.com/solidusio/solidus/discussions/new?category=troubleshooting) your issue.

## Feedback

We appreciate any feedback on the alpha version of Solidus Admin! Please comment on a [GitHub discussion](https://github.com/solidusio/solidus/discussions/new?category=new-admin-ui-ux-feedback) or [open an issue](https://github.com/solidusio/solidus/issues/new?assignees=&labels=type:bug,changelog:solidus_admin&projects=&template=bug_report.md&title=) if you find any bugs. Thanks!!

**Comment by tvdeyen** [comment]
Great work! This looks amazing. Will try it out in the next couple of days and will try to provide (valuable) feedback.

First thing I noticed in the video above is that the icon used for the quick filter is a "sort list" icon. Any particular reason why you have not used a search icon instead?

**Reply by waiting-for-dev** [reply]
Thanks, @tvdeyen! :heart:

Not sure that conveys sorting, as that's the "filter-line" icon from remix: https://remixicon.com/icon/filter-3-line :thinking:

**Reply by waiting-for-dev** [reply]
Other than that, it was the one given to us in the design, but of course we can talk to change it if we feel its confusing!

**Comment by peterberkenbosch** [comment]
Excited about this! Playing with it a little bit more next week, wanted to share first thing I ran into though. The Give Feedback link is 404-ing (https://solidus.io/feedback?category=solidus-admin).

**Reply by waiting-for-dev** [reply]
Thanks for reporting! The link is fixed now :rocket:

**Comment by zachfeldman** [comment]
Awesome can't wait to try it!

We tried putting it at `/new_admin` but it seems like some of the paths are maybe hardcoded to `/admin`. Could you all make sure it can be hosted at other URLs just so we could deploy it alongside existing admin? Thank you!

**Reply by waiting-for-dev** [reply]
Hey, @zachfeldman, could you be more precise about where you found that issue? :pray:

You should have the following after installing the new admin:

```ruby
  mount SolidusAdmin::Engine, at: '/admin', constraints: ->(req) {
    req.cookies['solidus_admin'] != 'false'
  }
```
Also, consider that if you mount it in a namespace different from the old backend one, the transparent fallback system won't work.

**Reply by zachfeldman** [reply]
Ah that makes sense. I tried putting it at just /admin and now I'm hitting this error, maybe because of how we have localization set up in our project?
![image](https://github.com/solidusio/solidus/assets/1437644/0c28de95-1b1d-4a47-a2b9-6b40e25c2f83)

**Reply by waiting-for-dev** [reply]
Thanks for reporting, @zachfeldman. https://github.com/solidusio/solidus/pull/5353 will fix that issue. Yeah, you probably have [raise_on_missing_translation](https://guides.rubyonrails.org/configuring.html#config-i18n-raise-on-missing-translations) set to `true`. We'll backport to the alpha branch once the PR is merged.

**Reply by waiting-for-dev** [reply]
Update: it's been already backported to the `admin/alpha/v1` branch. Please, update your bundle and try once more :pray:

**Reply by zachfeldman** [reply]
Nice thanks @waiting-for-dev . It is now loading, but I'm not seeing the new look on the Products admin. Maybe @elia could take a look at my open PR on our project to get a sense for what's wrong? I tagged him there. Appreciate all of your help.

**Reply by waiting-for-dev** [reply]
That's weird... To confirm, did you try playing with the left sidebar toggle switch, right?

A couple of ideas come to my mind for debugging it:

- Make sure you have the `mount SolidusAdmin::Engine` line **before** the `mount Spree::Core::Engine` one.
- Could you confirm the value of the `solidus_admin` cookie every time you click on the toggler?

By the way, Elia is enjoying some days off, so feel free to contact me for anything.

**Reply by zachfeldman** [reply]
I can't tell you why, but I created a new branch from @filippoliverani 's 4.1 branch and now it's working locally - awesome!

I'm trying to get a Review App up on Heroku so I can show the internal team here at @FrameworkComputer, but `bin/rake assets:precompile` fails on Heroku build with the following error:
```
       Running: rake assets:precompile
       
       Rebuilding...
       
       Done in 626ms.
       Specified input file /tmp/build_1582f9c9/app/assets/stylesheets/application.tailwind.css does not exist.
       rake aborted!
       Command failed with exit 9: /tmp/build_1582f9c9/vendor/bundle/ruby/3.1.0/gems/tailwindcss-rails-2.0.30-x86_64-linux/exe/x86_64-linux/tailwindcss
       /tmp/build_1582f9c9/vendor/bundle/ruby/3.1.0/gems/tailwindcss-rails-2.0.30-x86_64-linux/lib/tasks/build.rake:7:in `system'
       /tmp/build_1582f9c9/vendor/bundle/ruby/3.1.0/gems/tailwindcss-rails-2.0.30-x86_64-linux/lib/tasks/build.rake:7:in `block (2 levels) in <main>'
       /tmp/build_1582f9c9/vendor/bundle/ruby/3.1.0/gems/sentry-ruby-5.10.0/lib/sentry/rake.rb:26:in `execute'
       Tasks: TOP => assets:precompile => tailwindcss:build
       (See full trace by running task with --trace)

 !
 !     Precompiling assets failed.
 !
 !     Push rejected, failed to compile Ruby app.
 !     Push failed
```

I can file an issue in the project if you'd like.

**Reply by waiting-for-dev** [reply]
Thanks for reporting, @zachfeldman. It looks like deploying fails because the build directory is no longer there. We'll investigate how to work around that.
